io/GDALWriter.cpp

```diff
@@ -321,6 +326,21 @@ void GDALWriter::doneFile()
         throwError(raster.errorMsg());
 
     getMetadata().addList("filename", m_filename);
+
+    std::vector<std::string> gdalitems = Utils::split(m_GDAL_metadata, ',');
+    for (auto& v: gdalitems)
+    {
+
+        std::vector<std::string> item = Utils::split(v, '=');
+        std::string name = item[0];
+        std::string value = item[1];
```

Comment:
Would it be reasonable to handle `=` characters in the `value`?  In particular I'm thinking of base64 data e.g. `foo=ABCD==,bar=WXYZ==`.  As long as keys may not contain `=` which certainly seems to be the case, this is unambiguous since we've already split on commas, and seems simple enough to implement.

Reply:
It's GDAL that uses `=` as its split character, so I was just following the pattern there.

Reply:
This is UB for a specification without a right-hand side (e.g. `foo=,bar=`), right?  Maybe something of the form `std::string value = item.size() > 1 ? item[1] : ""`.

Reply:
> It's GDAL that uses = as its split character, so I was just following the pattern there.

Yeah that's fine - I'm saying that we could make sure we're using the *first* `=` as the split character.  As implemented, something like `expression=a+b=c` will store `expression` -> `a+b` in the GDAL metadata, since we seem to be discarding portions of the `value` following any `=` chars.

Reply:
Patches graciously accepted.

Reply:
I think something like this would solve both of these issues, although I haven't compiled it.
```
// If no "=" character is found, then the name is the entire token and the value is an empty string.
const std::size_t pos = item.find_first_of("=");
const std::string name = item.substr(0, pos);
const std::string value = pos != std::string::npos ? item.substr(pos + 1) : "";
raster.addMetadata(name, value);
```

Reply:
You can just check that `split` returns two things, no? Error otherwise.

Reply:
But why disallow `=` in the value, is what I was getting at.  Seems arbitrary.

Reply:
Because it's most likely that someone is just doing something stupid. IMO simple is good unless proved otherwise.